src/charm.py

```diff
@@ -1,88 +1,114 @@
 #!/usr/bin/env python3
-# Copyright 2023 nikos
+# Copyright 2023 Canonical Ltd.
 # See LICENSE file for licensing details.
 #
 # Learn more at: https://juju.is/docs/sdk
 
+"""A Juju charm for Identity Platform Login UI."""
 import logging
 
+from charms.observability_libs.v0.kubernetes_service_patch import KubernetesServicePatch
 from charms.traefik_k8s.v1.ingress import (
+    IngressPerAppReadyEvent,
     IngressPerAppRequirer,
+    IngressPerAppRevokedEvent,
 )
-from charms.observability_libs.v0.kubernetes_service_patch import KubernetesServicePatch
-from ops.charm import CharmBase
+from ops.charm import CharmBase, ConfigChangedEvent, HookEvent, WorkloadEvent
 from ops.main import main
-from ops.model import ActiveStatus, MaintenanceStatus
+from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus, WaitingStatus
+from ops.pebble import ChangeError, Layer
 
-# Log messages can be retrieved using juju debug-log
-logger = logging.getLogger(__name__)
+APPLICATION_PORT = "8080"
 
-VALID_LOG_LEVELS = ["info", "debug", "warning", "error", "critical"]
 
+logger = logging.getLogger(__name__)
 
-class KratosUiOperatorCharm(CharmBase):
-    """Charm the service."""
+
+class IdentityPlatformLoginUiOperatorCharm(CharmBase):
+    """Charmed Identity Platform Login UI."""
 
     def __init__(self, *args):
+        """Initialize Charm."""
         super().__init__(*args)
-        self._container_name = "kratos-ui"
+        self._container_name = "login-ui"
+        self._container = self.unit.get_container(self._container_name)
+
         self.service_patcher = KubernetesServicePatch(
-            self, [("api", 3000)]
+            self, [("identity-platform-login-ui", int(APPLICATION_PORT))]
         )
```

Comment:
Can we use open-port here?

Reply:
@nsklikas I suggested to go back to using `KubernetesServicePatch`. The reason is, [juju open-port](https://ops.readthedocs.io/en/latest/#ops.model.Unit.open_port) doesn't actually open the port unless you expose the charm first. Other charms in IAM stack don't need to be exposed with a command, so I think it's better to keep consistency and use the same here.

Reply:
Very well, let's leave this for now